patroni/dcs/__init__.py

```diff
@@ -318,8 +318,14 @@ def tags(self) -> Dict[str, Any]:
 
     @property
     def nofailover(self) -> bool:
-        """The value for ``nofailover`` in :attr:`Member`.tags`` if defined, otherwise ``False``."""
-        return self.tags.get('nofailover', False)
+        """The value for ``nofailover`` in :attr:`Member`.tags`` if defined. Otherwise, returns True
+        if ``failover_priority`` in :attr:`Member`.tags`` is non-positive, else defaults to ``False``."""
+        from_tags = self.tags.get('nofailover', None)
+        if from_tags is not None:
+            # Value of `nofailover` takes precedence over `failover_priority`
+            return bool(from_tags)
+        fp = parse_int(self.tags.get('failover_priority', 1))
+        return fp <= 0 if fp is not None else False
```

Comment:
I have the feeling we are duplicating this logic, and I wonder if is there a better way of handling that?